### ISSUE TITLE
Mysql stops after provisioning

### ISSUE THREAD

**Issue by keithdoyle9** [laravel/homestead#612]
Please note that the Homestead issue tracker is reserved for bug reports and enhancements. We are not always able to debug Vagrant, Provider or Operating System issues but will do our best to help. Thank you!

## Versions
  - Vagrant: v1.9.5
  - Provider: VMWare Workstation Pro v12.5.7
  - Homestead: v5.4.0

## Host operating system
Windows 7 Enterprise

## Homestead.yaml
```yaml
---
ip: "192.168.10.10"
memory: 2048
cpus: 2
provider: vmware_workstation
mariadb: true

authorize: C:/Users/kdoyle/.ssh/id_rsa.pub

keys:
    - C:/Users/kdoyle/.ssh/id_rsa

folders:
    - map: C:/Users/kdoyle/code
      to: /home/vagrant/projects

sites:
    - map: advisorbriefcase.seic.dev
      to: /home/vagrant/projects/advisorbriefcase-seic-com
    - map: trackingapp.dev
      to: /home/vagrant/projects/trackingapp/public
      type: laravel
    - map: seic.dev
      to: /home/vagrant/projects/seic-com

databases:
    - homestead
    - trackingapp
    - advbc

# blackfire:
#     - id: foo
#       token: bar
#       client-id: foo
#       client-token: bar

# ports:
#     - send: 50000
#       to: 5000
#     - send: 7777
#       to: 777
#       protocol: udp
```

## Vagrant up output
https://gist.github.com/keithdoyle9/43f847088dabef1bc8f185389b9a779f

## Expected behavior
MySQL service should be running and available

## Actual behavior
Can't connect error

## Steps to reproduce
1. Run vagrant up
2. Note the error at the end of the output about unable to connect
3. ssh into homestead
4. run `mysql` and note same error

**Comment by svpernova09** [comment]
Thanks for the bug report.

Do you have anything in `after.sh`? If so can you share it here.

**Comment by svpernova09** [comment]
Ah! I see it.

The key to this is you're running `vagrant up --provision` and the output is telling you Maria has already been installed: https://gist.github.com/keithdoyle9/43f847088dabef1bc8f185389b9a779f#file-gistfile1-txt-L61

My advice is to `vagrant destroy && vagrant up` to reset the entire box. If you don't want to destroy you can try running `rm -rf /home/vagrant/.maria` and then `vagrant up --provision` again. (Not 100% that will fix it, but worth a shot, if not, destroy & up will get it fixed)

**Comment by keithdoyle9** [comment]
It happens even after destroying and bringing up a brand new VM. Same issues as my colleague. We just recently upgraded to v5.4.0

**Comment by svpernova09** [comment]
Can you post a gist of the output of:

```
vagrant destroy -f && vagrant up
```

I haven't been able to recreate the issue.

**Comment by svpernova09** [comment]
Also run `vagrant box outdated` and make sure you're running the newest version of the base box.

**Comment by keithdoyle9** [comment]
It's v.2.2.0, we upgraded the same time we upgraded to v5.4.0

Gist of destroy & up

https://gist.github.com/keithdoyle9/81c0a1126fb106569ea73e3ca7e88a89

**Comment by keithdoyle9** [comment]
It works for now. But this is what happens each time. Can't find a root cause on what would cause it to stop.

**Comment by svpernova09** [comment]
That gist output looks good. don't see any errors. Nothing has changed regarding Maria in releases lately.

Glad you got it working. If you find a repeatable scenario that causes the bug please let us know.

**Comment by keithdoyle9** [comment]
Haven't done anything, left VM running. Exited out and ssh'd back in to find that mysql doesn't work again.

`vagrant@homestead ~ $ mysql
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2 "No such file or directory")
vagrant@homestead ~ $`

**Comment by svpernova09** [comment]
Are you suspending or sleeping your machine?

**Comment by keithdoyle9** [comment]
Nope, all I did was exit out and ssh back in.

**Comment by svpernova09** [comment]
I have no idea, check the logs to see what's happening?

**Comment by TopFuel** [comment]
Hi, I think I ran into this issue some time ago, it has todo with AppArmor and MariaDB.
If I remember correctly I found the fix here: https://askubuntu.com/q/750604
On the [approved answer](https://askubuntu.com/a/753280) the person explains what's happening and gives fix, but it didn't work for me so I tried the fix on the [answer by plutocrat](https://askubuntu.com/a/888715) and it worked, so I added this to the after.sh:
```
if [ -f /home/vagrant/.maria ]
then
  sudo ln -s /etc/apparmor.d/usr.sbin.mysqld /etc/apparmor.d/disable/
  sudo /etc/init.d/apparmor reload
  sudo service mysql restart
fi
```
And it's working with no problem for some time now, I was going to post this here on github but I forgot.
I don't remember having this issue with the normal MySQL, so it seems like it's only happening when using MariaDB.
I don't know if it's the same issue, but it seems so.
